lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -263,12 +278,19 @@ def configure(conf)
       # Functions.
       @running_cloudfunctions = false
 
-      # Set labels, etc. based on the config
+      # Set up the MonitoredResource, labels, etc. based on the config
       case @platform
       when Platform::GCE
-        @service_name = COMPUTE_SERVICE
+        @resource.type = COMPUTE_RESOURCE_TYPE
+        # TODO: introduce a new MonitoredResource-centric configuration and
+        # deprecate subservice-name; for now, translate known uses.
         if @subservice_name
-          @service_name = @subservice_name
+          # TODO: what should we do if we encounter an unknown value?
+          if @subservice_name == DATAFLOW_SERVICE
+            @resource.type = DATAFLOW_RESOURCE_TYPE
+          elsif @subservice_name == ML_SERVICE
+            @resource.type = ML_RESOURCE_TYPE
```

Comment:
Yeah, `subservice_name` is a config we support to override subservice detection (each subservice more or less corresponds to a specific resource type). We would need to support the same behavior after V2 migration.

For the other resource types, I was reading the discussion from the original V2 migration thread. It sounds to me: The initial goal is to support the services we currently support, without requiring users to make changes to their configuration. We should eventually enable users to specify the resource type and any required labels explicitly in the configuration. Yet that is not something we are planning to do for this round. And we can only fully deprecate the concept of "service name" then.